### PR TITLE
Add model

### PR DESCRIPTION
**Pull request by tienhxh**
Anfis: Epoch 159/200, Train Loss: 108.5372, Val Loss: 154.2652, compared with
h0 Construct: Epoch 200/200, Train Loss: 79.5573, Val Loss: 102.9604.
SARIMAX: bad result

### CODE REVIEW

File: models/regression/simple_mlp.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        self.hidden_dim = hidden_dim
+        self.output_dim = output_dim
+        super(BasicMLP, self).__init__(input_dim)
+
+        self.model = self._build_network()
+
+    def _build_network(self) -> nn.Module:
+        return MLPNetwork(self.input_dim, self.hidden_dim, self.output_dim)
+
+    def forward(self, batch: torch.Tensor) -> torch.Tensor:
```
Comment (nguyenhongson1996) [inline models/regression/simple_mlp.py:45]: Add comment for the size of the input tensor.

File: models/regression/simple_mlp.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        x = self.relu(x)
+        x = self.fc2(x)
+        x = self.relu(x)
+        return self.fc3(x)
+
+class BasicMLP(BaseModel):
+    def __init__(self, input_dim: int, hidden_dim: int, output_dim: int):
+        """
+        Basic MLP model.
+        :param input_dim: Number of input features.
```
Comment (nguyenhongson1996) [inline models/regression/simple_mlp.py:32]: Add more detail about the expected value of this parameter

File: models/regression/simple_mlp.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        :param hidden_dim: Number of neurons in the hidden layer.
+        :param output_dim: Number of output features.
```
Comment (nguyenhongson1996) [inline models/regression/simple_mlp.py:34]: What is expected size of output?

File: models/regression/simple_transformer.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        """
+        Transformer-based regression model.
+        :param input_dim: Number of input features.
+        :param model_dim: Hidden dimension of the Transformer.
+        :param output_dim: Number of output features.
+        :param num_heads: Number of attention heads.
+        :param num_layers: Number of encoder layers.
+        """
+        super(TransformerNetwork, self).__init__()
+        self.embedding = nn.Linear(input_dim, model_dim)
```
Comment (nguyenhongson1996) [inline models/regression/simple_transformer.py:17]: Why do we need embedding?

File: models/regression/simple_transformer.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        self.encoder_layer = nn.TransformerEncoderLayer(d_model=model_dim, nhead=num_heads)
+        self.transformer = nn.TransformerEncoder(self.encoder_layer, num_layers=num_layers)
+        self.fc = nn.Linear(model_dim, output_dim)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.embedding(x)  # Transform input into a vector of size model_dim
```
Comment (nguyenhongson1996) [inline models/regression/simple_transformer.py:23]: Can we not use the model_dim as same as the input dim? What is the reason for introducing model_dim?

File: models/regression/simple_transformer.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        x = x.unsqueeze(1)  # Add sequence length dimension = 1 (batch_size, seq_len=1, model_dim)
```
Comment (nguyenhongson1996) [inline models/regression/simple_transformer.py:24]: if the sequence length is 1 then what is the reason for using Transformer model?

**Review by nguyenhongson1996 — CHANGES_REQUESTED** [review]
Why do we use transformer with sequence length = 0? Where is SARIMAX and Anfis?

File: models/regression/simple_transformer.py (hunk 5 above)
Comment (tienhxh) [reply models/regression/simple_transformer.py:23]: We can set model_dim = input_dim, but input_dim merely reflects the number of features in the input data,
model_dim allows the model to learn a representation in a different space, potentially more complex or simpler than the input space.

File: models/regression/simple_transformer.py (hunk 4 above)
Comment (tienhxh) [reply models/regression/simple_transformer.py:17]: Transformers perform efficiently when all vectors have the same dimension, d_model (here, model_dim).

Comment (nguyenhongson1996) [reply models/regression/simple_transformer.py:17]: That is not the answer. What does embedding layer stand for?

File: tests/test_anfis.py
Comment (tienhxh) [inline tests/test_anfis.py:1]: Anfis: Epoch 159/200, Train Loss: 108.5372, Val Loss: 154.2652, compared with 
h0 Construct: Epoch 200/200, Train Loss: 79.5573, Val Loss: 102.9604

File: tests/test_sarimax.py
Comment (tienhxh) [inline tests/test_sarimax.py:1]: i used grid search to choose the best parameters but it was a bad result: 
Train Loss (MSE) = 309.527821
Validation Loss (MSE) = 274.594403

File: tests/test_anfis.py
Comment (levi-lb) [reply tests/test_anfis.py:1]: Add these information to PR description

File: models/regression/simple_anfis.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+class ANFISNetwork(nn.Module):
+    def __init__(self, input_dim: int, num_rules: int, output_dim: int):
+        super(ANFISNetwork, self).__init__()
+        self.input_dim = input_dim
+        self.num_rules = num_rules
+        self.output_dim = output_dim
+
+        self.membership_layer = nn.Linear(input_dim, num_rules)
+        self.rule_layer = nn.Linear(num_rules, output_dim)
```
Comment (nguyenhongson1996) [inline models/regression/simple_anfis.py:17]: Then this model contains just two linear layers? The output will be a linear translation from the input, which explains the low accuracy of the model. Anfis contains 5 layers, where are other layers? https://www.dca.ufrn.br/~meneghet/FTP/anfis%2093.pdf

**Review by nguyenhongson1996 — CHANGES_REQUESTED** [review]
Looks like ANFIS implementation is incorrect.
